plugins/lookup/mergelist.py

```diff
@@ -0,0 +1,76 @@
+# (c) 2020, Jose Angel Munoz <josea.munoz(at)gmail.com>
+# (c) 2020 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    lookup: mergelist
+    author: Jose Angel Munoz <josea.munoz@gmail.com>
+    version_added: "2.10"
+    short_description: merges lists from a given array key
+    description:
+      - This lookup returns the contents of two or more merged lists using a common key.
+    options:
+      _terms:
+        description: list(s) to merge
+        required: True
+    notes:
+      - if key is not defined, "name" will be the default
+      - use query if you want to return a list when a single value is given.
+'''
+
+EXAMPLES = """
+  - name: Merges list1 and list2 with 'name' as key
+    debug:
+      msg: "{{ query('mergelist', list1, list2) }}"
+    vars:
+      list1:
+        - name: myname01
+          param01: myparam01
+        - name: myname02
+          param01: myparam02
+      list2:
+        - name: myname01
+          param01: myparam03
+        - name: myname02
+          param02: myparam04
+        - name: myname03
+          param03: myparam03
+
+  - name: Merges list1 and list2 with 'id' as key
+    debug:
+      msg: "{{ query('mergelist', 'id', list1, list2) }}"
```

Comment:
You should use keyword arguments for such parameters:
```suggestion
      msg: "{{ query('mergelist', list1, list2, key='id') }}"
```

Reply:
Good one. Thanks!